akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -370,14 +372,17 @@ private[stream] final class VirtualProcessor[T] extends Processor[T, T] {
           try {
             subscriptionStatus.set(s)
             tryOnSubscribe(s, sub)
+            if (isCancelled) throw canNotSubscribeTheSameSubscriberMultipleTimesException
             sub.closeLatch() // allow onNext only now
             terminationStatus.getAndSet(Allowed) match {
               case null       ⇒ // nothing happened yet
               case Completed  ⇒ tryOnComplete(s)
               case Failed(ex) ⇒ tryOnError(s, ex)
               case Allowed    ⇒ // all good
+
             }
           } catch {
+            case ex @ canNotSubscribeTheSameSubscriberMultipleTimesException ⇒ throw ex
```

Comment:
This is not correct, this will catch all exceptions.
canNotSubscribeTheSameSubscriberMultipleTimesException is like any bind variable in the match.
It would have had to be `canNotSubscribeTheSameSubscriberMultipleTimesException` or starting with upper case.

Perhaps:

      } catch {
         case NonFatal(ex) ⇒ if (isCancelled) throw ex else sub.cancel()
      }